### PR TITLE
Implement cache service

### PR DESCRIPTION
**Pull request by dlloyd1-ch**
Add main function to initialise and start service.

Added configuration for client and redis services.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -4,9 +4,39 @@ import "github.com/ian-kent/gofigure"
...
+	return ServiceConfig{c}
+}
+
+// ServiceConfig wraps Config to implement service.Config.
+type ServiceConfig struct {
+	Config
+}
+
+func (cfg ServiceConfig) BindAddr() string {
+	return cfg.BindAddr()
```
Comment (dlgroves-ch) [inline config/config.go]: Causes a stack overflow error on my machine

File: config/config.go [hunk 2]
```diff
@@ -4,9 +4,39 @@ import "github.com/ian-kent/gofigure"
...
+}
+
+func (cfg ServiceConfig) CertFile() string {
+	return cfg.CertFile()
```
Comment (dlgroves-ch) [inline config/config.go]: Causes a stack overflow error on my machine

**Review by dlgroves-ch — CHANGES_REQUESTED** [review]

File: service/service.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+			cfg.poolSize,
+			cfg.expiryInSeconds,
+		),
+		s.topic,
+		logger.NewLogger())
+	return s
+}
+
+func (s *CacheService) Start() {
+	go s.broker.Run()
```
Comment (dlgroves-ch) [inline service/service.go:85]: We also need to start the client here:
```golang
go s.client.Connect()
```

**Review by dlgroves-ch — COMMENTED** [review]
The request handler also needs to append a newline to the outgoing message else it won't appear in the response.

File: config/config.go (hunk 1 above)
Comment (dlloyd1-ch) [reply config/config.go]: Fixed

File: config/config.go (hunk 2 above)
Comment (dlloyd1-ch) [reply config/config.go]: Fixed

**Review by dlgroves-ch — APPROVED** [review]

**Pull request merged**